### PR TITLE
[16.0][MIG] account_journal_lock_date: Migration to 16.0

### PR DESCRIPTION
**Pull request by RodrigoBM**
#1472

### CODE REVIEW

**Comment by alexis-via** [comment]
I made a PR to improve this PR https://github.com/factorlibre/account-financial-tools/pull/7
It is a forward port of the fix that just landed in v14 : https://github.com/OCA/account-financial-tools/pull/1459

**Comment by alexis-via** [comment]
When my PR will be merged, I'll approve this PR.

**Comment by RodrigoBM** [comment]
@alexis-via PR merged. Thanks

**Comment by RodrigoBM** [comment]
@alexis-via  the tests fail.
The [odoo code](https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_move.py#L2202) only checks the lock date in the posted state.
If we want to keep the logic of the tests, I think it is necessary to check this in the creation.

```
    @api.model_create_multi
    def create(self, vals_list):
        res = super().create(vals_list)
        res._check_fiscalyear_lock_date()
        return res
```
what do you think?

**Review by alexis-via — APPROVED** [review]
code review

**Comment by RodrigoBM** [comment]
@alexis-via  he borrado la comprobación de error en la creación  de account.move en los test, ya que los test estaban fallando.
En odoo la comprobacion de fecha bloqueo solo se realiza al pasar al estado posted, por lo cual ahora se permitira crear elementos en fecha de bloqueo.

**Comment by RodrigoBM** [comment]
@alexis-via  I have deleted the error check on the creation of account.move in the tests, as the tests were failing.
In odoo the lock date check is only done when moving to the posted state, so now it will be allowed to create items on lock date.

File: account_journal_lock_date/tests/test_journal_lock_date.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    def test_journal_lock_date(self):
+        self.env.user.write({"groups_id": [(3, self.ref("base.group_system"))]})
+        self.env.user.write(
+            {"groups_id": [(3, self.ref("account.group_account_manager"))]}
+        )
+        self.assertFalse(self.env.user.has_group("account.group_account_manager"))
+
+        # Test that the move cannot be written, or cancelled
+        with self.assertRaises(UserError):
```
Comment (astirpe) [inline account_journal_lock_date/tests/test_journal_lock_date.py]: @RodrigoBM do you mind to replace all the occurrences of `with self.assertRaises(UserError):` with:

```suggestion
        with self.assertRaisesRegex(UserError, ".*prior to and inclusive of the lock date.*"):
```

This way we ensure that the correct error is raised.

**Review by astirpe — CHANGES_REQUESTED** [review]

File: account_journal_lock_date/tests/test_journal_lock_date.py (hunk 1 above)
Comment (RodrigoBM) [reply account_journal_lock_date/tests/test_journal_lock_date.py]: done

**Review by astirpe — APPROVED** [review]

**Comment by RodrigoBM** [comment]
@alexis-via can tou merge it? please

**Comment by RodrigoBM** [comment]
@rafaelbn can tou merge it?

**Review by alexis-via — APPROVED** [review]

**Pull request merged**